search.go

```diff
@@ -196,7 +205,7 @@ func (c *SearchCriteria) Format() (fields []interface{}) {
 		fields = append(fields, "BCC", c.Bcc)
 	}
 	if !c.Before.IsZero() {
-		fields = append(fields, "BEFORE", FormatDate(c.Before))
+		fields = append(fields, "BEFORE", c.Before.Format(DateLayout))
```

Comment:
Either we choose to format dates here and we remove types `Date` and `DateTime`, either we choose to always use them. But I'd prefer not to mix both.